lib/remote/objectqueryhandler.cpp

```diff
@@ -189,6 +190,9 @@ bool ObjectQueryHandler::HandleRequest(
 		joinAttrs.insert(field.Name);
 	}
 
+	std::unordered_map<Type*, std::pair<Expression::Ptr, bool>> evaluatedTypes;
+	std::unordered_map<Object*, bool> joinedRelations;
```

Comment:
Could use better variable names, the current names don't really describe what they store. Something like `typePermissions` and `objectAccessAllowed` would be better.

Also, in the first map, I'd make it `std::pair<bool, Expression::Ptr>` as that's the order the values are used. If the bool is false, the filter isn't even considered.